backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/CpuPinningHelper.java

```diff
@@ -100,26 +102,93 @@ private static List<Integer> createRange(int start, int end) {
     }
 
     /**
-     * Provides a CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
+     * Provides a dedicated CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
      *
      * @param vdsCpuUnits a list of VdsCpuUnit
      * @return a map of CPU pinning
      */
-    public static Map<Integer, Integer> createCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
+    public static Map<Integer, Integer> createDedicatedCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
         return IntStream.range(0, vdsCpuUnits.size())
                 .boxed()
                 .collect(Collectors.toMap(Function.identity(), i -> vdsCpuUnits.get(i).getCpu()));
     }
 
+    /**
+     * Provides a CPU pinning map based on the VdsCpuUnit list and the CPU pinning policy. The pinning will be based on the list order.
+     * @param vdsCpuUnits a list of VdsCpuUnit
+     * @param cpuPinningPolicy the CPU pinning policy
+     * @return a map of CPU pinning
+     */
+    public static Map<Integer, Integer> createExclusiveCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits, CpuPinningPolicy cpuPinningPolicy) {
+        switch(cpuPinningPolicy) {
+            case DEDICATED:
+                return createDedicatedCpuPinningMap(vdsCpuUnits);
+            case ISOLATE_THREADS:
+                return createIsolatedThreadsCpuPinningMap(vdsCpuUnits);
+        }
+        return null;
+    }
+
+    /**
+     * Provides an isolate threads CPU pinning map based on the VdsCpuUnit list. The pinning will be based on the list order.
+     *
+     * @param vdsCpuUnits a list of VdsCpuUnit
+     * @return a map of CPU pinning
+     */
+    private static Map<Integer, Integer> createIsolatedThreadsCpuPinningMap(List<VdsCpuUnit> vdsCpuUnits) {
+        List<Integer> socketsUsed = vdsCpuUnits.stream().map(VdsCpuUnit::getSocket).distinct().collect(Collectors.toList());
```

Comment:
can we come up with a better name than socketsUsed? it's not cleared what they are used for - you mean that we should use here?